column_buffer_go18.go

```diff
@@ -178,6 +181,35 @@ func writeRowsFuncOfOptional(t reflect.Type, schema *Schema, path columnPath, wr
 	}
 }
 
+func writeRowsFuncOfTime(_ reflect.Type, schema *Schema, path columnPath) writeRowsFunc {
+	t := reflect.TypeOf(int64(0))
+	elemSize := uintptr(t.Size())
+	writeRows := writeRowsFuncOf(t, schema, path)
+
+	return func(columns []ColumnBuffer, rows sparse.Array, levels columnLevels) error {
+		if rows.Len() == 0 {
+			return writeRows(columns, rows, levels)
+		}
+
+		for i := 0; i < rows.Len(); i++ {
+			p := (*time.Time)(rows.Index(i))
+			a := sparse.Array{}
+			if p != nil {
+				var nanos int64
+				if !p.IsZero() {
```

Comment:
Why is this check necessary?

Reply:
When working on my PR, it was hard for me to know when zero and nil checks are necessary versus when they aren't.  How can we tell whether or not they'd be necessary given the function signatures for the `funcOf` patterns?

Reply:
I guess Go doesn't help much here, generally I would say that pointers are never intended to be `nil` except when they are paired with an error in the return values of a function (in which case we usually test the error). If that's not the case it should be documented on exported APIs.

Panics for accessing nil pointers seem like a good enough safeguard then.

In this particular case, the `sparse.(*Array).Index` method returns a pointer to the element at the given index, so it cannot ever be nil.